kafka-companion-frontend/src/app/util/array-sort.pipe.ts

```diff
@@ -0,0 +1,45 @@
+import {Injectable, Pipe, PipeTransform} from "@angular/core";
+import {TopicMetadata} from "../topics/topic-metadata";
+
+@Injectable({
+  providedIn: 'root'
+})
+@Pipe({
+  name: 'arraySort'
+})
+export class ArraySortPipe implements PipeTransform {
+  transform(value: any[], key: string, order: string): any {
+    return [...value].sort((a, b) => this.compareValues(a, b, key, order));
+  }
+
+  private compareValues(a: any, b: any, key: string, order: string) {
+    if (!a.hasOwnProperty(key) || !b.hasOwnProperty(key)) {
+      return 0;
+    }
+
+    const varA = this.getValue(a, key);
+    const varB = this.getValue(b, key);
+
+    if (a.group === b.group) {
+      return this.compareWithOrder(varA, varB, order);
+    } else if (a.group === TopicMetadata.GROUP_FAVOURITES) {
+      return -1;
+    } else if (b.group === TopicMetadata.GROUP_FAVOURITES) {
+      return 1;
+    }
+  }
+
+  private compareWithOrder(varA: any, varB: any, order: string) {
```

Comment:
Mam pewne wątpliwości - to zadziała tylko jeśli mamy string bez polski znaków

Reply:
Przetestowałem przypadek z polskimi znakami i jak dla mnie działa poprawnie

Reply:
Zmieniłem na localeCompare. Powinno być lepiej